docs/how-to-guides/rook-ceph-storage.md

```diff
@@ -0,0 +1,293 @@
+
+```bash
+lokoctl cluster apply -v --skip-components
+```
+
+## Deploy `rook`
+
+#### Config
+
+Create a file named `storage.lokocfg` with the following contents:
+
+```tf
+component "rook" {
+  node_selector = {
+    "storage.lokomotive.io" = "ceph"
+  }
+
+  toleration {
+    key      = "storage.lokomotive.io"
+    operator = "Equal"
+    value    = "ceph"
+    effect   = "NoSchedule"
+  }
+
+  agent_toleration_key    = "storage.lokomotive.io"
+  agent_toleration_effect = "NoSchedule"
+
+  discover_toleration_key    = "storage.lokomotive.io"
+  discover_toleration_effect = "NoSchedule"
+}
+```
+
+- `node_selector` should match the `labels` variable provided in the `worker_pool`.
+- `toleration` should match the `taints` variable mentioned in the `worker_pool`.
+- `agent_toleration_key` and `discover_toleration_key` should match the `key` of the `taints` variable provided in the `worker_pool`.
+- `agent_toleration_effect` and `discover_toleration_effect` should match the `effect` of the `taints` variable provided in the `worker_pool`.
+
+For information about all the available configuration options for the `rook` component, visit the component's [configuration reference](../configuration-reference/components/rook.md).
+
+#### Deploy the component
+
+Execute the following command to deploy the `rook` component:
+
+```bash
+lokoctl component apply rook
+```
+
+Verify the operator pod in the `rook` namespace is in the `Running` state (this may take a few minutes):
+
+```console
+$ kubectl -n rook get pods -l app=rook-ceph-operator
+NAME                                  READY   STATUS    RESTARTS   AGE
+rook-ceph-operator-76d8687f95-6knf8   1/1     Running   0          2m
+```
+
+## Deploy `rook-ceph`
+
+#### Config
+
+Add following contents to the previously created file `storage.lokocfg`:
+
+```tf
+component "rook-ceph" {
+  monitor_count = 3
+
+  node_affinity {
+    key      = "storage.lokomotive.io"
+    operator = "Exists"
+  }
+
+  toleration {
+    key      = "storage.lokomotive.io"
+    operator = "Equal"
+    value    = "ceph"
+    effect   = "NoSchedule"
+  }
+
+  storage_class {
+    enable = true
+  }
+}
+```
+
+- `monitor_count` should be an odd number greater than three and not higher than the `count` variable of workers in the `worker_pool`.
+- `node_affinity` should match the `labels` variable provided in the `worker_pool`.
+- `toleration` should match the `taints` variable provided in the `worker_pool`.
+
+For information about all the available configuration options for the `rook-ceph` component, visit the component's [configuration reference](../configuration-reference/components/rook-ceph.md).
+
+#### Deploy the component
+
+Execute the following command to deploy the `rook-ceph` component:
+
+```bash
+lokoctl component apply rook-ceph
+```
+
+Verify the [OSD](https://docs.ceph.com/docs/master/glossary/#term-ceph-osd-daemon) pods in the `rook` namespace are in the `Running` state (this may take a few minutes):
```

Comment:
The actual verification should be done by inspecting ceph cluster object, as it contains a state field, updated by operator.

Reply:
I wasn't aware of such a field, I have used OSDs coming up as a check to verify that now everything is converged.

Reply:
Ah, my bad. It seems Rook docs actually recommends using the toolbox to verify cluster state: https://rook.io/docs/rook/v1.4/ceph-quickstart.html.  I'm not sure which we should recommend then.

Reply:
I think checking OSDs are up and running has been proved to be reliable thus far.

Reply:
Fair enough, thanks.